### PR TITLE
[BugFix] Fix wrong state of 'isExecuteInOneTablet' when it comes to agg or analytic

### LINKED ISSUE #19725: [Bug] Wrong result of agg
### Steps to reproduce the behavior (Required)

```sql
CREATE TABLE testtable (
history_type tinyint(4) NOT NULL ,
history_id int(11) NOT NULL ,
history_time datetime NOT NULL,
history_category varchar(120) NOT NULL,
created_by varchar(10) NOT NULL,
created_time datetime NOT NULL
) ENGINE=OLAP
DUPLICATE KEY(history_type, history_id, history_time, history_category)
DISTRIBUTED BY HASH(history_type) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "true"
);


INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2018-12-31 00:00:00', 'aaaa','yliu68', '2023-02-07 18:44:14');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2019-12-31 00:00:00', 'bbbb', 'yliu68', '2023-02-07 18:44:14');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2020-12-31 00:00:00', 'cccc', 'yliu68', '2023-02-07 18:44:14');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2021-12-31 00:00:00', 'dddd', 'yliu68', '2023-02-07 18:44:14');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2022-12-31 00:00:00', 'eeee', 'yliu68', '2023-02-07 18:44:14');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('7', '162', '2022-12-31 00:00:00', 'ffff', 'yliu68', '2023-02-23 15:29:47');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('10', '162', '2023-12-31 00:00:00', 'gggg', 'hwu16', '2023-02-20 09:53:11');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('10', '162', '2023-12-31 00:00:00', 'hhhh', 'hwan2', '2023-02-20 16:26:06');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('10', '162', '2023-12-31 00:00:00', 'iiii', 'yma4', '2023-02-20 18:11:19');
INSERT INTO testtable (history_type, history_id, history_time, history_category, created_by, created_time) VALUES ('10', '162', '2023-12-31 00:00:00', 'jjjj', 'yma4', '2023-02-20 18:16:31');


SELECT history_id AS supplier_id
FROM (
    SELECT
        history_id,
        history_time,
        dense_rank () over (PARTITION BY history_id, history_time ORDER BY created_time DESC ) AS rk
    FROM testtable
    WHERE
        history_type = 7
        AND history_id = 162
) temp
WHERE temp.rk = 1
GROUP BY history_id
```

### Expected behavior (Required)

```
+-------------+
| supplier_id |
+-------------+
| 162         |
+-------------+
```

### Real behavior (Required)

```
+-------------+
| supplier_id |
+-------------+
| 162         |
+-------------+
| 162         |
+-------------+
| 162         |
+-------------+
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19725

## Problem Summary(Required) ：

The derivation of `isExecuteInOneTablet` is wrong when it comes to agg or analytic node that changes the data distribution. 

Besides, analytic node can also benefit from one tablet optimization.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 62 / 62 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitAggregateRule.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/LogicalProperty.java|46|46|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/LogicalProperty.java [hunk 1]
```diff
@@ -72,66 +83,147 @@ public void derive(ExpressionContext expressionContext) {
         isExecuteInOneTablet = op.accept(new OneTabletExecutorVisitor(), expressionContext);
     }
 
-    static class OneTabletExecutorVisitor extends OperatorVisitor<Boolean, ExpressionContext> {
+    public static final class OneTabletMeta {
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/LogicalProperty.java]: Good choice.

**Review by Seaven — APPROVED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Review by ZiheLiu — APPROVED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 69 / 69 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitAggregateRule.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ExpressionContext.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/LogicalProperty.java|52|52|100.00%|[]|

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.0 branch-2.5 branch-2.4 branch-2.3

**Comment by liuyehcf** [comment]
https://github.com/Mergifyio backport branch-2.3

**Comment by liuyehcf** [comment]
Fixed in 2.3.12, 2.4.5, 2.5.5

**Pull request merged**